extensions/amp-list/0.1/amp-list.js

```diff
@@ -943,6 +943,29 @@ export class AmpList extends AMP.BaseElement {
 
   }
 
+  /**
+   * @param {boolean=} opt_refresh
+   * @return {!Promise<!JsonObject|!Array<JsonObject>>}
+   * @private
+   */
+  prepareAndSendFetch_(opt_refresh = false) {
+    // If cross-origin="amp-viewer-auth-token-via-post" attribute is present,
+    // the viewer will make a remote xhr POST request with an auth token in the
+    // request body. Requires amp-viewer-assistance extension for auth token.
+    const crossOriginAttr = this.element.getAttribute('cross-origin');
+    if (crossOriginAttr &&
+        crossOriginAttr.indexOf('amp-viewer-auth-token-via-post') >= 0) {
+      return Services.viewerAssistanceForDocOrNull(this.win)
+          .then(viewerAssistance => {
+            devAssert(viewerAssistance,
+                'Viewer Assistance service can not be found');
+            return viewerAssistance.getIdTokenPromise();
+          })
```

Comment:
Can we de-dupe this code snippet (checking `crossorigin` and getting the token) by moving it into xhr-utils.js?

```js
/** 
 * @param {!Element}
 * @return {!Promise<string|undefined>} 
 */
function getViewerAuthTokenIfAvailable(element) {}
```

Then this could just be:
```js
return getViewerAuthTokenIfAvailable(element).then(token => this.fetch_(opt_refresh, token));
```

Reply:
Done.